### ISSUE TITLE
Warnings in verbose mode

### ISSUE THREAD

**Issue by dduugg** [sorbet/sorbet#3782]
#### Input

```shell
ruby --verbose -e 'require "sorbet-runtime"'
```

#### Observed output

```
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/private/abstract/data.rb:15: warning: instance variable @opus_abstract__abstract_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/props/private/apply_default.rb:17: warning: method redefined; discarding old initialize
/path/to/sorbet-runtime-0.5.6145/lib/types/private/abstract/declare.rb:37: warning: previous definition of initialize was here
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/private/abstract/data.rb:15: warning: instance variable @opus_abstract__abstract_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/props/private/apply_default.rb:71: warning: method redefined; discarding old initialize
/path/to/sorbet-runtime-0.5.6145/lib/types/private/abstract/declare.rb:37: warning: previous definition of initialize was here
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
/path/to/sorbet-runtime-0.5.6145/lib/types/types/simple.rb:44: warning: instance variable @__as_sorbet_simple_type not initialized
```

#### Expected behavior

Requiring the gem in verbose mode should not emit warnings (this is the case when i replace `sorbet-runtime` with a number of other popular gems, such as `rails`, `rake`, `rack`, `rubocop`, `rspec`).

This is a bummer when using the default `rspec` config (via `rspec --init`, which enables verbose mode by setting `config.warnings = true`), making for noisy tests if one wants to track warnings in a project that uses `sorbet-runtime`.

**Comment by jez** [comment]
I think it's probably just a matter of pushing this PR to completion

#3207

I haven't looked at how much work that is.

**Comment by paracycle** [comment]
This PR pre-dates mine https://github.com/sorbet/sorbet/pull/3167 and it was also solving the method redefinition errors.

On the other hand, the longer we wait, the less we will need this to be fixed: https://bugs.ruby-lang.org/issues/17055#change-89059 😂

**Comment by jez** [comment]
It looks like this is fixed now, maybe in #5482 but I haven't confirmed it was that PR for sure.
